### PR TITLE
chore: migrate `cloudavenue_(catalogs/catalog_vapp_template)` to superschema

### PR DESCRIPTION
**Pull request by azrod**
### Description of your changes

If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### CODE REVIEW

**Comment by valentin-dubreuil** [comment]
## :bangbang: Acceptance Test failed 

=== RUN   TestAccCatalogsDataSource
--- PASS: TestAccCatalogsDataSource (14.87s)
PASS
ok  	github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests/catalog	14.877s

**Comment by valentin-dubreuil** [comment]
## :bangbang: Acceptance Test failed 

=== RUN   TestAccCatalogVappTemplateDataSource
    catalog_vapp_template_datasource_test.go:24: Test validation error: TestStep 1/2 validation error: TestStep missing Config or ImportState or RefreshState
--- FAIL: TestAccCatalogVappTemplateDataSource (0.00s)
FAIL
FAIL	github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests/catalog	0.007s
FAIL

**Comment by valentin-dubreuil** [comment]
## :bangbang: Acceptance Test failed 

=== RUN   TestAccCatalogVappTemplateDataSource
--- PASS: TestAccCatalogVappTemplateDataSource (13.79s)
PASS
ok  	github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests/catalog	13.801s

File: internal/provider/catalog/catalog_vapp_template_datasource.go [hunk 1]
```diff
@@ -87,78 +82,68 @@ func (d *vAppTemplateDataSource) Read(ctx context.Context, req datasource.ReadRe
...
-			if vAppTemplate.Name == state.TemplateName.ValueString() {
-				state.TemplateID = types.StringValue(vAppTemplate.ID)
-				href = vAppTemplate.HREF
-				break
-			}
-		}
+	for _, vAppTemplate := range vAppTemplates {
+		if (state.TemplateID.IsKnown() && vAppTemplate.ID == state.TemplateID.Get()) || (state.TemplateName.IsKnown() && vAppTemplate.Name == state.TemplateName.Get()) {
+			// field ID in vAppTemplate attribute is always empty. ID exist in HREF attribute.
+			vappTemplateID := uuid.Normalize(uuid.VAPPTemplate, vAppTemplate.HREF[len(vAppTemplate.HREF)-36:])
```
Comment (dmicheneau) [inline internal/provider/catalog/catalog_vapp_template_datasource.go:104]: -36 ? an explication

File: internal/provider/catalog/catalogs_datasource.go [hunk 2]
```diff
@@ -159,6 +167,8 @@ func (d *catalogsDataSource) Read(ctx context.Context, req datasource.ReadReques
 		CatalogsName: cn,
 	}
 
+	tflog.Info(ctx, pretty.Sprint(updatedState))
```
Comment (dmicheneau) [inline internal/provider/catalog/catalogs_datasource.go:170]: delete this debug print

File: internal/provider/catalog/catalogs_datasource.go [hunk 3]
```diff
@@ -7,6 +7,8 @@ import (
 	"net/url"
 	"sort"
 
+	"github.com/kr/pretty"
```
Comment (dmicheneau) [inline internal/provider/catalog/catalogs_datasource.go:10]: delete and go mod tidy

File: internal/provider/catalog/catalogs_schema.go [hunk 4]
```diff
@@ -1,29 +1,113 @@
...
-		Attributes: map[string]schema.Attribute{
-			"id": schema.StringAttribute{
-				Computed: true,
+func catalogsSuperSchema(_ context.Context) superschema.Schema {
+	return superschema.Schema{
+		Common: superschema.SchemaDetails{
+			MarkdownDescription: "The catalogs datasource show the details of all the catalogs.",
+		},
+		Attributes: map[string]superschema.Attribute{
+			"id": superschema.StringAttribute{
```
Comment (dmicheneau) [inline internal/provider/catalog/catalogs_schema.go:19]: why you don't use supertypes ?

**Review by dmicheneau — CHANGES_REQUESTED** [review]

File: internal/provider/catalog/catalog_vapp_template_datasource.go (hunk 1 above)
Comment (azrod) [reply internal/provider/catalog/catalog_vapp_template_datasource.go:104]: UUID containt 36 characters

File: internal/provider/catalog/catalogs_schema.go (hunk 4 above)
Comment (azrod) [reply internal/provider/catalog/catalogs_schema.go:19]: Because it is not planned to re-write the datasource

**Review by dmicheneau — APPROVED** [review]

**Pull request merged**